site/src/paradox/internals/Coders.md

```diff
@@ -41,31 +91,69 @@ While it saves a lot of work, it also has a few drawbacks:
 
 ## Scio `0.7.0` and above
 
-In Scio `0.7.0` and above, the Scala compiler will try to find the correct instance of `Coder` at compile time.
-In most cases, the compiler should be able to either directly find a proper `Coder` implementation, or derive one automatically.
+In Scio `0.7.0` and above, the Scala compiler will try to find the correct instance of `Coder` at compile time. In most cases, the compiler should be able to either directly find a proper `Coder` implementation, or derive one automatically.
+
+### Scio `Coder` vs Beam `Coder`
+
+Both Scio and Beam define a class called `Coder`. For the most part when writing a job, you will be interacting with `com.spotify.scio.coders.Coder`.
+
+Scio `Coder` and its implementations simply form an [ADT](https://en.wikipedia.org/wiki/Algebraic_data_type) where each implementation is a building block that covers one of the possible cases:
+
+- `Beam`: a simple wrapper around a Beam Coder
+- `Disjunction`: Represent a Coder that makes a choice between different possible implementations. It is for example used to serialize ADTs and `Either`
+- `Record`: A Coder for record-like structures like case classes and tuples.
+- `Transform`: A Coder implemented by "transforming" another Coder.
+- `Fallback`: A default `Coder`. Used when there is no better option.
 
-Please note that Scio wraps Beam coders in its own `Coder` definition: `com.spotify.scio.coders.Coder`
+There is also a "special" coder called `KVCoder`. It is a specific coder for Key-Value pairs. Internally Beam treats [KV](https://beam.apache.org/releases/javadoc/2.20.0/org/apache/beam/sdk/values/KV.html) differently from other types so Scio needs to do the same.
```

Comment:
I think we could use the `@javadoc[KV](org.apache.beam.sdk.values.KV)` or did it not work in this case? advantage is that points to whatever beam version we are using

Reply:
I did not know `@javadoc` existed :/